shuffle/cli/src/shared.rs

```diff
@@ -103,6 +107,86 @@ pub fn get_shuffle_dir() -> PathBuf {
     BaseDirs::new().unwrap().home_dir().join(".shuffle")
 }
 
+pub struct DevApiClient {
+    client: Client,
+}
+
+// Client that will make GET and POST requests based off of Dev API
+impl DevApiClient {
+    pub fn new(client: Client) -> Result<Self> {
+        Ok(Self { client })
+    }
+
+    pub async fn get_account_modules(
+        &self,
+        network: &Url,
+        address: AccountAddress,
+    ) -> Result<Response> {
+        let path =
+            network.join(format!("accounts/{}/modules", address.to_hex_literal()).as_str())?;
+        Ok(self.client.get(path.as_str()).send().await?)
+    }
+
+    pub async fn get_transactions_with_hash(&self, hash: &str, network: &Url) -> Result<Response> {
+        let path = network.join(format!("transactions/{}", hash).as_str())?;
+        Ok(self.client.get(path.as_str()).send().await?)
+    }
+
+    pub async fn post_transactions(&self, network: &Url, txn_bytes: Vec<u8>) -> Result<Response> {
+        let path = network.join("transactions")?;
+        Ok(self
+            .client
+            .post(path.as_str())
+            .header("Content-Type", "application/vnd.bcs+signed_transaction")
+            .body(txn_bytes)
+            .send()
+            .await?)
+    }
+
+    pub async fn get_account_sequence_number(
+        &self,
+        network: &Url,
+        address: AccountAddress,
+    ) -> Result<u64> {
+        let path =
+            network.join(format!("accounts/{}/resources", address.to_hex_literal()).as_str())?;
+        let resp = self.client.get(path.as_str()).send().await?;
```

Comment:
you probably should extract above 3 lines into `get_account_resources`, then this function calls to it to get resource

Reply:
great idea!